tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -144,26 +175,39 @@ constexpr bool test() {
             double v3;
         };
 
+        struct S7 {
+            int f1() {
+                return 0;
+            }
+        };
+
+        struct NS : S1, S2 {}; // Non-standard layout
+
+        ASSERT(is_corresponding_member(&S1::v1, &S::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S::v2));
         ASSERT(is_corresponding_member(&S1::v1, &S1::v1));
         ASSERT(is_corresponding_member(&S1::v2, &S1::v2));
-        ASSERT(!is_corresponding_member(&S1::v1, &S1::v2));
-        ASSERT(!is_corresponding_member(&S1::v2, &S1::v1));
-        ASSERT(is_corresponding_member(&S1::v1, &S2::v1));
-        ASSERT(is_corresponding_member(&S1::v2, &S2::v2));
+        ASSERT(is_corresponding_member(&S1::v1, &S2::w1));
+        ASSERT(is_corresponding_member(&S1::v2, &S2::w2));
         ASSERT(is_corresponding_member(&S1::v1, &S3::v1));
         ASSERT(is_corresponding_member(&S1::v2, &S3::v2));
         ASSERT(is_corresponding_member(&S5::v1, &S6::v1));
         ASSERT(is_corresponding_member(&S5::v2, &S6::v2));
 
-        ASSERT(!is_corresponding_member(&S1::v1, &S2::v2));
-        ASSERT(!is_corresponding_member(&S1::v2, &S2::v1));
+        ASSERT(!is_corresponding_member(&S1::v1, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S1::v1));
+        ASSERT(!is_corresponding_member(&S1::v2, &S2::w1));
         ASSERT(!is_corresponding_member(&S1::v1, &S4::v1));
         ASSERT(!is_corresponding_member(&S1::v2, &S4::v2));
         ASSERT(!is_corresponding_member(&S3::v1, &S4::v1));
         ASSERT(!is_corresponding_member(&S3::v2, &S4::v2));
         ASSERT(!is_corresponding_member(&S5::v1, &S6::v2));
         ASSERT(!is_corresponding_member(&S5::v2, &S6::v1));
         ASSERT(!is_corresponding_member(&S5::v3, &S6::v3));
+        ASSERT(!is_corresponding_member<NS, NS>(&NS::v1, &NS::w1));
+        ASSERT(!is_corresponding_member(&S7::f1, &S7::f1));
+        ASSERT(!is_corresponding_member<S1, S2, int, int>(nullptr, nullptr));
+        ASSERT(!is_corresponding_member<S1, S2, int, int>(&S1::v1, nullptr));

```

Comment:
I think it would be slightly better to use `static_cast` to form null pointers of the desired types, allowing template argument deduction to be used normally. That will make the remaining explicit template arguments (for `NS`) stand out as being necessary. I'll push a change for this (and the `nullptr` below).